src/validator/FindPathValidator.cpp

```diff
@@ -142,5 +145,82 @@ GetNeighbors::EdgeProps FindPathValidator::buildEdgeKey(bool reverse) {
                 });
     return edgeProps;
 }
+
+Status FindPathValidator::allPairPaths() {
+    auto* bodyStart = StartNode::make(qctx_);
+    auto* passThrough = PassThroughNode::make(qctx_, bodyStart);
+
+    auto* forward = allPaths(passThrough, from_, false);
+    VLOG(1) << "forward: " << forward->outputVar();
+
+    auto* backward = allPaths(passThrough, to_, true);
+    VLOG(1) << "backward: " << backward->outputVar();
+
+    auto* conjunct = ConjunctPath::make(
+        qctx_, forward, backward, ConjunctPath::PathKind::kAllPaths, steps_.steps);
+    conjunct->setLeftVar(forward->outputVar());
+    conjunct->setRightVar(backward->outputVar());
+    conjunct->setColNames({"_path"});
+
+    auto* loop = Loop::make(
+        qctx_, nullptr, conjunct, buildAllPathsLoopCondition(steps_.steps));
+
+    auto* dataCollect = DataCollect::make(
+        qctx_, loop, DataCollect::CollectKind::kAllPaths, {conjunct->outputVar()});
+    dataCollect->setColNames({"_path"});
+
+    root_ = dataCollect;
+    tail_ = loop;
+    return Status::OK();
+}
+
+PlanNode* FindPathValidator::allPaths(PlanNode* dep, const Starts& starts, bool reverse) {
+    std::string startVidsVar;
+    Expression* vids = nullptr;
+    buildConstantInput(starts, startVidsVar, vids);
+
+    DCHECK(!!vids);
+    auto* gn = GetNeighbors::make(qctx_, dep, space_.id);
+    gn->setSrc(vids);
+    gn->setEdgeProps(buildEdgeKey(reverse));
+    gn->setInputVar(startVidsVar);
+
+    auto* allPaths = ProduceAllPaths::make(qctx_, gn);
+    allPaths->setInputVar(gn->outputVar());
+    allPaths->setColNames({"_vid", "path"});
+    allPaths->setOutputVar(startVidsVar);
+
+    DataSet ds;
+    ds.colNames = {"_vid", "path"};
+    for (auto& vid : starts.vids) {
+        Row row;
+        row.values.emplace_back(vid);
+
+        List paths;
+        Path path;
+        path.src = Vertex(vid.getStr(), {});
```

Comment:
vid must string?

Reply:
Yes, string is the only option now for `Vertex`.